robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,222 @@
+import logging
+import re
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents ReportPortal
+
+    This holds the properties and functions to interact with Report Portal properties and launches
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
+
+    @property
+    def api_url(self):
+        """Super url of report portal
+        :returns: Base url for API request
+        """
+        return f'{self.rp_url}/api/v1/{self.rp_project}/'
+
+    @property
+    def headers(self):
+        """The headers for Report Portal Requests.
+        :returns: header for API request
+        """
+        return {'Authorization': f'Bearer {self.rp_api_key}'}
+
+    def launches(self, sat_version=None, launch_type='satellite6', latest=200):
```

Comment:
Don't mean to nitpick here, but two things about the `latest` kwarg:

1. Default of 200 seems very high for our intended use, is there a reason to default to so many launches?
2. The naming seems off, as we use the `latest` convention more like a boolean, to indicate we're getting the 'latest' version of something (latest rhel, latest satellite snap, etc).  Something like `count` would be more appropriate here I think, with the docblock explaining the count is considered from the latest chronological launch working backwards.

Reply:
Actually 200 was defined when we didn't have typed and versioned launches. It can be cut down max 30 now. Thanks for bringing that up. Good catch.